src/sentry/static/sentry/app/stores/groupStore.tsx

```diff
@@ -2,58 +2,114 @@ import isArray from 'lodash/isArray';
 import isUndefined from 'lodash/isUndefined';
 import Reflux from 'reflux';
 
+import {Activity, Group} from 'app/types';
 import GroupActions from 'app/actions/groupActions';
 import IndicatorStore from 'app/stores/indicatorStore';
 import {t} from 'app/locale';
 
 function showAlert(msg, type) {
-  IndicatorStore.add(msg, type, {
+  IndicatorStore.addMessage(msg, type, {
     duration: 4000,
   });
 }
 
+// TODO(ts) Type this any better.
+type Change = [string, string, any];
+
 class PendingChangeQueue {
-  constructor() {
-    this.changes = [];
-  }
+  changes: Change[] = [];
 
-  getForItem(itemId) {
-    return this.changes.filter(change => change[1] === itemId);
+  getForItem(itemId: string) {
+    return this.changes.filter((change: Change) => change[1] === itemId);
   }
 
-  push(changeId, itemId, data) {
+  push(changeId: string, itemId: string, data: any) {
     this.changes.push([changeId, itemId, data]);
   }
 
-  remove(changeId, itemId) {
+  remove(changeId: string, itemId?: string) {
     this.changes = this.changes.filter(
       change => change[0] !== changeId || change[1] !== itemId
     );
   }
 
-  forEach() {
-    this.changes.forEach.apply(this.changes, arguments);
+  forEach(...args: any[]) {
+    this.changes.forEach.apply(this.changes, args);
   }
 }
 
-const GroupStore = Reflux.createStore({
+type Internals = {
+  items: any[];
+  statuses: Record<string, Record<string, boolean>>;
+  pendingChanges: PendingChangeQueue;
+
+  indexOfActivity: (groupId: string, id: string) => number;
+  addActivity: (groupId: string, data: Activity, index?: number) => void;
+  updateActivity: (groupId: string, id: string, data: Partial<Activity>) => void;
+  removeActivity: (groupId: string, id: string) => number;
+};
+
+type GroupStoreInterface = Reflux.StoreDefinition & {
+  init: () => void;
+  reset: () => void;
+  loadInitialData: (items: Group[]) => void;
+  add: (items: Group[]) => void;
+  remove: (itemId: string) => void;
+  addStatus: (id: string, status: string) => void;
+  clearStatus: (id: string, status: string) => void;
+  hasStatus: (id: string, status: string) => boolean;
+  get: (id: string) => Group | undefined;
+  getAllItemIds: () => string[];
+  getAllItems: () => Group[];
+  onAssignTo: (changeId: string, itemId: string, data: any) => void;
```

Comment:
Following along from above

```suggestion
  onAssignTo: (...change: Change) => void;
```